orc8r/cloud/go/services/certifier/servicers/certifier.go

```diff
@@ -712,60 +718,166 @@ func (srv *CertifierServer) getPolicyDecisionFromTokenMany(tokens *certprotos.To
 	return &certprotos.GetPolicyDecisionResponse{Effect: finalEffect}, nil
 }
 
-func (srv *CertifierServer) getPolicyDecisionFromToken(token string, reqResource *certprotos.Resource) (certprotos.Effect, error) {
-	policy, err := srv.store.GetPolicy(token)
+func (srv *CertifierServer) getPolicyDecisionFromToken(ctx context.Context, token string, req *certprotos.Request) (certprotos.Effect, error) {
+	policyList, err := srv.store.GetPolicy(token)
 	if err != nil {
-		return certprotos.Effect_UNKNOWN, status.Errorf(codes.Internal, "failed to get policy from db, checking next policy: %v", err)
+		return certprotos.Effect_DENY, status.Errorf(codes.Internal, "failed to get policyList from db %v", err)
 	}
 	effect := certprotos.Effect_UNKNOWN
-	for _, policyResource := range policy.Resources.Resources {
-		actionEffect := checkAction(reqResource, policyResource)
-		if actionEffect == certprotos.Effect_DENY {
-			return certprotos.Effect_DENY, nil
-		}
-		resourceEffect := checkResource(reqResource, policyResource)
-		if resourceEffect == certprotos.Effect_DENY {
-			return certprotos.Effect_DENY, nil
-		}
+
+	// Networks are registered with tenants, hence any tenant scoped policies
+	// have additional policies for their networks.
+	tenantNetworkResource, err := getTenantPolicyNetworkResourceMany(ctx, policyList)
+	if err != nil {
+		return effect, nil
+	}
+	policyList.Policies = append(policyList.Policies, tenantNetworkResource...)
+
+	for _, policy := range policyList.Policies {
+		actionEffect := isActionAuthorized(req, policy)
+		resourceEffect := isResourceAuthorized(req, policy)
 		// The effects from both action and resource should match
-		// for the effect of the policy to apply to the request.
+		// for the effect of the policyList to apply to the request.
 		if actionEffect != certprotos.Effect_UNKNOWN &&
 			resourceEffect != certprotos.Effect_UNKNOWN &&
 			actionEffect == resourceEffect {
 			effect = actionEffect
 		}
+
+		if actionEffect == certprotos.Effect_DENY &&
+			resourceEffect == certprotos.Effect_DENY {
```

Comment:
1. Keep this on a single line for better skim-ability
2. I think this needs a comment. It's not immediately obvious why we would need these to *both* be "deny"